MultimodalQnA/ui/gradio/multimodalqna_ui_gradio.py

```diff
@@ -315,6 +315,64 @@ def ingest_image_gen_caption(filepath, request: gr.Request):
     return
 
 
+def ingest_image_with_text(filepath, text, request: gr.Request):
+    yield (gr.Textbox(visible=True, value="Please wait for your uploaded image to be ingested into the database..."))
+    verified_filepath = os.path.normpath(filepath)
+    if not verified_filepath.startswith(tmp_upload_folder):
+        print("Found malicious image file name!")
+        yield (
+            gr.Textbox(
+                visible=True,
+                value="Your uploaded image's file name has special characters that are not allowed. Please consider updating the file name!",
+            )
+        )
+        return
+    basename = os.path.basename(verified_filepath)
+    dest = os.path.join(static_dir, basename)
+    shutil.copy(verified_filepath, dest)
+    text_basename = "{}.txt".format(os.path.splitext(basename)[0])
+    text_dest = os.path.join(static_dir, text_basename)
+    with open(text_dest, "a") as file:
+        file.write(text)
+    print("Done copying uploaded files to static folder!")
+    headers = {
+        # 'Content-Type': 'multipart/form-data'
+    }
+    files = [
+        ('files', (basename, open(dest, "rb"))),
+        ('files', (text_basename, open(text_dest, "rb")))
+    ]
+    response = requests.post(dataprep_ingest_addr, headers=headers, files=files)
+    print(response.status_code)
+    os.remove(text_dest)
+    if response.status_code == 200:
+        response = response.json()
+        print(response)
+        yield (gr.Textbox(visible=True, value="Image ingestion is done. Saving your uploaded image..."))
+        time.sleep(2)
+        fn_no_ext = Path(dest).stem
+        if "file_id_maps" in response and fn_no_ext in response["file_id_maps"]:
+            new_dst = os.path.join(static_dir, response["file_id_maps"][fn_no_ext])
+            print(response["file_id_maps"][fn_no_ext])
+            os.rename(dest, new_dst)
+            yield (
+                gr.Textbox(
+                    visible=True,
+                    value="Congratulation! Your upload is done!\nClick the X button on the top right of the image upload box to upload another image.",
+                )
+            )
+            return
+    else:
+        yield (
+            gr.Textbox(
+                visible=True,
+                value="Something went wrong!\nPlease click the X button on the top right of the image upload box to reupload your image!",
```

Comment:
Is there more information that we can give on why it failed?

Reply:
There's probably a better way. I'll create a ticket for this since it would be a site-wide update.

Reply:
https://jira.devtools.intel.com/browse/AIZOO-3560